config/locales/server.en.yml

```diff
@@ -4018,8 +4018,12 @@ en:
     store_failure: "Failed to store upload #%{upload_id} for user #%{user_id}."
     file_missing: "Sorry, you must provide a file to upload."
     empty: "Sorry, but the file you provided is empty."
+    failed: "Sorry, but your upload failed. Please try again."
     png_to_jpg_conversion_failure_message: "An error happened when converting from PNG to JPG."
     optimize_failure_message: "An error occurred while optimizing the uploaded image."
+    download_failure: "Downloading the file from the external provider failed."
```

Comment:
I wonder if these are of much use to an end user? Most users will only want to know their upload failed and to try again, rather than if there's a download failure or checksum mismatch. Those are very advanced messages that I suspect will only confuse people. They might be good for admins, though.

Reply:
Yeah that is fair enough. When you say good for admins do you mean I should show these error messages to admin users and the generic "upload failed" one to everyone else?

Reply:
I was actually thinking of logging them instead rather than displaying them.

Reply:
Okay done!